include/aspect/material_model/utilities.h

```diff
@@ -268,6 +268,16 @@ namespace aspect
 
 
 
+      /**
+       * Used for calculating average elastic shear modulus and viscosity
+       */
+      template <int dim>
+      double calculate_average_vector (const std::vector<double> &composition,
```

Comment:
it doesn't calculate an "average vector" but it averages a vector into a double, right? Can you think of a better name? calculate_average_of_vector? average_vector?
And sorry, I realize you are just moving things around.